MonitoringSystem/DB/MonitoringDB.py

```diff
@@ -0,0 +1,171 @@
+########################################################################
+# $Id: $
+########################################################################
+
+"""
+It is a wrapper on top of Elasticsearch. It is used to manage the DIRAC monitoring types. 
+"""
+__RCSID__ = "$Id$"
+
+from DIRAC.Core.Base.ElasticDB import ElasticDB
+from DIRAC.MonitoringSystem.private.TypeLoader import TypeLoader
+from DIRAC import S_OK, S_ERROR, gConfig, gLogger
+
+########################################################################
+class MonitoringDB( ElasticDB ):
+
+  def __init__( self, name = 'Monitoring/MonitoringDB', readOnly = False ):
+    ElasticDB.__init__( self, 'MonitoringDB', name )
+    self.__readonly = readOnly
+    self.__documents = {}
+    self.__loadIndexes()
+       
+  def __loadIndexes( self ):
+    """
+    It loads all monitoring indexes and types.
+    """
+    retVal = gConfig.getSections( "/DIRAC/Setups" )
+    if not retVal[ 'OK' ]:
+      return S_ERROR( "Can't get a list of setups: %s" % retVal[ 'Message' ] )
+    
+    setupsList = retVal[ 'Value' ]
+    objectsLoaded = TypeLoader().getTypes()
+
+    # Load the files
+    for pythonClassName in sorted( objectsLoaded ):
+      typeClass = objectsLoaded[ pythonClassName ]
+      for setup in setupsList:
+        indexName = "%s_%s" % ( setup.lower(), typeClass().getIndex() )
+        doc_type = typeClass().getDocType() 
+        mapping = typeClass().getMapping()
+        monfields = typeClass().getMonitoringFields()
+        self.__documents[doc_type] = {"indexName": indexName, "mapping":mapping, 'monitoringFields':monfields}
+        if self.__readonly:
+          gLogger.info( "Read only mode is okay" )
+        else:
+          self.registerType( doc_type, indexName, mapping )
+  
+  def getIndexName( self, typeName ):
+    """
+    :param tyeName it is a string. doc_type and type name is equivalent
+    """
+    indexName = None
+    
+    if typeName in self.__documents:
+      indexName = self.__documents.get( typeName ).get( "indexName", None )
+      
+    if indexName:
+      return S_OK( indexName )
+    else:
+      return S_ERROR( "The index of %s not found!" % typeName )
+  
+  def registerType( self, mtype, index, mapping ):
+    """
+    It register the type and index, if does not exists
+    :param str type: Type of the documents
+    :param str index: name of the index
+    """ 
+    
+    all_index = "%s-*" % index
+    
+    if self.isExists( all_index ):  
+      indexes = self.getIndexes()
+      if len( indexes ) > 0:
```

Comment:
I do not like if indexes. when I write len(indexes) I know the return value of the method is iterable. It is easier to understand the code